### PR TITLE
chore(Github Action|CI): add Github Action for testing

### PR DESCRIPTION
**Pull request by MichelDiz**
Reusing existing test scripts to bring CI into the repository.

- - -
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph-js-http/66)

### CODE REVIEW

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs
```
Comment (joshua-goldstein) [inline .github/workflows/node.js.yml]: we don't need comments at top of workflow

File: .github/workflows/node.js.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs
+
+name: Node.js CI
```
Comment (joshua-goldstein) [inline .github/workflows/node.js.yml]: change name to ci-dgraph-js-http-tests

File: .github/workflows/node.js.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs
+
+name: Node.js CI
+
+on:
+  push:
+    branches: [ "master" ]
```
Comment (joshua-goldstein) [inline .github/workflows/node.js.yml]: don't need to use matrix here

File: .github/workflows/node.js.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs
+
+name: Node.js CI
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
```
Comment (joshua-goldstein) [inline .github/workflows/node.js.yml]: don't need matrix here

File: .github/workflows/node.js.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs
+
+name: Node.js CI
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+
```
Comment (joshua-goldstein) [inline .github/workflows/node.js.yml]: we might want to add a cron job @skrdgraph  thoughts?

File: .github/workflows/node.js.yml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
```
Comment (joshua-goldstein) [inline .github/workflows/node.js.yml]: change to ubuntu-20.04, remove extra space

File: .github/workflows/node.js.yml [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+    strategy:
+      matrix:
+        node-version: [14.x, 16.x, 18.x, 19.x]
```
Comment (joshua-goldstein) [inline .github/workflows/node.js.yml]: these are a lot of node versions, is there a particular reason you chose these?

File: .github/workflows/node.js.yml [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    - uses: actions/checkout@v3
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ matrix.node-version }}
+        cache: 'npm'
+    - name: Install Dependencies
+      run: |
+        chmod +x ./scripts/build.sh
+        chmod +x ./scripts/install_dgraph.sh
```
Comment (joshua-goldstein) [inline .github/workflows/node.js.yml]: remove this line, shouldn't be necessary

File: .github/workflows/node.js.yml [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+    steps:
+    - uses: actions/checkout@v3
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ matrix.node-version }}
+        cache: 'npm'
+    - name: Install Dependencies
+      run: |
+        chmod +x ./scripts/build.sh
```
Comment (joshua-goldstein) [inline .github/workflows/node.js.yml]: remove this line and invoke build.sh with bash

**Review by joshua-goldstein — CHANGES_REQUESTED** [review]

**Comment by joshua-goldstein** [comment]
Looking good at first glance, just some cosmetic things.  Try to make the CI here look like other CI files as much as possible, e.g. https://github.com/dgraph-io/dgraph/blob/main/.github/workflows/ci-dgraph-tests.yml

File: .github/workflows/node.js.yml (hunk 7 above)
Comment (MichelDiz) [reply .github/workflows/node.js.yml]: Same answer as in the dgraph-js repo.

**Review by joshua-goldstein — APPROVED** [review]

**Pull request merged**